lib/netsuite/configuration.rb

```diff
@@ -7,11 +7,16 @@ def reset!
 
       clear_wsdl_cache
 
+      Thread.main[:netsuite_gem_multi_tenant] = false
       attributes.clear
     end
 
     def attributes
-      Thread.current[:netsuite_gem_attributes] ||= {}
+      if Thread.main[:netsuite_gem_multi_tenant]
+        Thread.current[:netsuite_gem_attributes] ||= (Thread.main[:netsuite_gem_attributes] || {}).dup
+      else
+        Thread.main[:netsuite_gem_attributes] ||= {}
+      end
```

Comment:
My thought here is something like:

```suggestion
      if @multi_tenant
        Thread.current[:netsuite_gem_attributes] ||= {}
      else
        @attributes
      end
```

i.e. we don't want to copy config if we are multi-tenant, it's up to each process to set the NS config before processing.